x-pack/plugins/fleet/server/services/files/index.test.ts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClientMock } from '@kbn/core/server/mocks';
+import { elasticsearchServiceMock } from '@kbn/core/server/mocks';
+
+import { ES_SEARCH_LIMIT } from '../../../common/constants';
+import {
+  FILE_STORAGE_DATA_INDEX,
+  FILE_STORAGE_METADATA_INDEX,
+} from '../../constants/fleet_es_assets';
+
+import { fileIdsWithoutChunksByIndex, getFilesByStatus, updateFilesStatus } from '.';
+
+const ENDPOINT_FILE_METADATA_INDEX = '.fleet-endpoint-files';
+const ENDPOINT_FILE_INDEX = '.fleet-endpoint-file-data';
+
+describe('files service', () => {
+  let esClientMock: ElasticsearchClientMock;
+
+  beforeEach(() => {
+    esClientMock = elasticsearchServiceMock.createElasticsearchClient();
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('#getFilesByStatus()', () => {
+    it('should return expected values', async () => {
+      const status = 'READY';
+      esClientMock.search.mockResolvedValueOnce({
+        took: 5,
+        timed_out: false,
+        _shards: {
+          total: 1,
+          successful: 1,
+          skipped: 0,
+          failed: 0,
+        },
+        hits: {
+          hits: [
+            {
+              _index: ENDPOINT_FILE_METADATA_INDEX,
```

Comment:
silly nitpick, but since we're in fleet-land now, should we change these references to be "about" agent instead of endpoint?

Reply:
tbh I just don't know the index names 😅. might be bad search-fu on my end but I couldn't spot any references to other `.fleet-*-files` type indices. happy to update if anyone can point me in the right direction.

Reply:
I have used .fleet-agent-files in my [pr](https://github.com/elastic/kibana/pull/142369/files), probably doesn't matter too much for the unit test.